### PR TITLE
Cleanup and divide in functions

### PR DESCRIPTION
**Pull request by cesar-lopez-mar**

### CODE REVIEW

File: src/napoleon_navigation_rosnode.cpp [hunk 1]
```diff
@@ -618,77 +553,884 @@ int main(int argc, char** argv)
...
-    std::ofstream myfile;
-    //myfile.open ("/simdata/ropod_" + get_date() +".txt");
-    myfile.open ("/home/cesar/Documents/simdata/ropod_" + get_date() +".txt");
-    myfile << "time" << "\t" << "tictoc" << "\t" << "state" << "\t" << "task counter" << "\t" << "tube width" << "\t" <<
-            "phi des" << "\t" << "v ropod odom" << "\t"<< "v ropod cmd" << "\t" << "x ropod" << "\t" << "y ropod" << "\t" <<
-            "theta ropod" << "\t" "x obs" << "\t" << "y obs" << "\t" << "theta obs" << "\t" << "obs width" << "\t" <<
-            "obs depth" << "\t" << "obs vx" << "\t" << "obs vy" << "\t" << "des accel" <<"\n";
 
-    std::clock_t start_loop;
+void overtakeStateMachine()
```
Comment (Lakshadeep) [inline src/napoleon_navigation_rosnode.cpp:610]: It's better to move all these functions to separate ROS independent class.

**Comment by Lakshadeep** [comment]
I guess this can be merged for now. But, we will need further refactoring for adding monitors and switching between perception and motion control behaviours.

@cesar-lopez-mar is this tested?

Planner interface is also ready and tested, so can be merged. But now we will have lots of merge conflicts, so I suggest we merge this first and then we try to merge the planner interface.

**Comment by cesar-lopez-mar** [comment]
Indeed that was the idea from the beginning. We go step by step, with this changes at least is more esay to read what the main loop is doing.   Btw, yes I already tested it in simulation.
We can first merge this and then the interface.

File: src/napoleon_navigation_rosnode.cpp (hunk 1 above)
Comment (cesar-lopez-mar) [reply src/napoleon_navigation_rosnode.cpp:610]: Indeed, this is just a first step. Due to the limited amount of time I have to spend on this task, first I first divided the main in these functions to improve the readability, which eventually would help continuing the refactoring by someone else. As agreed on previous discussions, further refactoring is needed.

**Pull request merged**